services/frontendhost/frontend.go

```diff
@@ -238,11 +238,21 @@ func convertCreateDestRequestToInternal(createRequest *c.CreateDestinationReques
 func convertUpdateDestRequestToInternal(updateRequest *c.UpdateDestinationRequest, destUUID string) *shared.UpdateDestinationRequest {
 	internalUpdateRequest := shared.NewUpdateDestinationRequest()
 	internalUpdateRequest.DestinationUUID = common.StringPtr(destUUID)
-	internalUpdateRequest.Status = common.InternalDestinationStatusPtr(shared.DestinationStatus(updateRequest.GetStatus()))
```

Comment:
this function is causing the bug. Basically if the field is not specified in CLI, we still incorrectly created a pointer, then in metadata_cassandra it'll treat this field as 'specified' and update metadata using these default values.